services/apps/src/apps/clients/notifications/app_sharing.clj

```diff
@@ -0,0 +1,73 @@
+(ns apps.clients.notifications.app-sharing
+  (:require [clojure.string :as string]))
+
+(def grouping-threshold 10)
+
+(def notification-type "apps")
+(def share-action "share")
+(def unshare-action "unshare")
+
+(def sharer-success-formats
+  {:grouped   {:singular "%5$d app has been %1$sd with %3$s."
+               :plural   "%5$d apps have been %1$sd with %3$s."}
+   :ungrouped {:singular "The following app has been %1$sd with %3$s: %4$s"
+               :plural   "The following apps have been %1$sd with %3$s: %4$s"}})
+
+(def sharee-success-formats
+  {:grouped   {:singular "%2$s has %1$sd %5$d app with you."
+               :plural   "%2$s has %1$sd %5$d apps with you."}
+   :ungrouped {:singular "%2$s has %1$sd the following app with you: %4$s"
+               :plural   "%2$s has %1$sd the following apps with you: %4$s"}})
+
+(def failure-formats
+  {:grouped   {:singular "%5$d app could not be %1$sd with %3$s."
+               :plural   "%5$d apps could not be %1$sd with %3$s."}
+   :ungrouped {:singular "The following app could not be %1$sd with %3$s: %4$s"
+               :plural   "The following apps could not be %1$sd with %3$s: %4$s"}})
```

Comment:
Meh, yeah, a better formatting tool might be nice, but I still though that this was better than piecing the strings together using `(str)`